src/BiometryService/BiometryService.Android.cs

```diff
@@ -0,0 +1,426 @@
+	{
+		if (_logger.IsEnabled(LogLevel.Debug))
+		{
+			_logger.LogDebug($"Encrypting the fingerprint for the key '{keyName}'.");
+		}
+
+		await ValidateBiometryCapabilities(ct);
+
+		var crypto = CreateCryptoObject(keyName);
+		var result = await AuthenticateBiometry(ct, crypto);
+		var valueToEncrypt = Encoding.UTF8.GetBytes(value);
+		var encryptedData = result.CryptoObject.Cipher.DoFinal(valueToEncrypt);
+		var iv = result.CryptoObject.Cipher.GetIV();
+
+		var bytes = new byte[iv.Length + encryptedData.Length];
+		iv.CopyTo(bytes, 0);
+		encryptedData.CopyTo(bytes, iv.Length);
+
+		if (_logger.IsEnabled(LogLevel.Information))
+		{
+			_logger.LogInformation($"Succcessfully encrypted the fingerprint for the key'{keyName}'.");
+		}
+
+		var encodedData = Base64.EncodeToString(bytes, Base64Flags.NoWrap);
+		var sharedpref = _applicationContext.GetSharedPreferences(PREFERENCE_NAME, FileCreationMode.Private);
+		sharedpref.Edit().PutString(keyName, encodedData).Apply();
+	}
+
+	/// <inheritdoc/>
+	public async Task<string> Decrypt(CancellationToken ct, string keyName)
+	{
+		if (_logger.IsEnabled(LogLevel.Debug))
+		{
+			_logger.LogDebug($"Decrypting the fingerprint for the key '{keyName}'.");
+		}
+
+		await ValidateBiometryCapabilities(ct);
+
+		var sharedpref = _applicationContext.GetSharedPreferences(PREFERENCE_NAME, FileCreationMode.Private);
+		var storedData = sharedpref.GetString(keyName, null);
+
+		if (storedData == null)
+		{
+			throw new BiometryException(BiometryExceptionReason.KeyInvalidated, "Encrypted values could not be found. It may have been removed.");
+		}
+
+		byte[] data = Base64.Decode(storedData, Base64Flags.NoWrap);
+
+		var iv = new byte[16];
+		Array.ConstrainedCopy(
+			data,
+			0,
+			iv,
+			0,
+			16
+		);
+
+		var buffer = new byte[data.Length - 16];
+		Array.ConstrainedCopy(
+			data,
+			16,
+			buffer,
+			0,
+			data.Length - 16
+		);
+
+		var crypto = GetCryptoObject(keyName, iv);
+		var result = await AuthenticateBiometry(ct, crypto);
+		var decryptedData = result.CryptoObject.Cipher.DoFinal(buffer);
+
+		if (_logger.IsEnabled(LogLevel.Information))
+		{
+			_logger.LogInformation($"Succcessfully decrypted the fingerprint for the key '{keyName}'.");
+		}
+
+		return Encoding.ASCII.GetString(decryptedData);
+	}
+
+	/// <inheritdoc/>
+	public void Remove(string keyName)
+	{
+		var sharedpref = _applicationContext.GetSharedPreferences(PREFERENCE_NAME, FileCreationMode.Private);
+		sharedpref.Edit().Remove(keyName).Apply();
+	}
+
+	private async Task<BiometricPrompt.AuthenticationResult> AuthenticateBiometry(CancellationToken ct, BiometricPrompt.CryptoObject crypto = null)
+	{
+		if (_logger.IsEnabled(LogLevel.Debug))
+		{
+			_logger.LogDebug($"Start authenticating the user biometry.");
+		}
+
+		var authenticateCode = 0;
+		if (Android.OS.Build.VERSION.SdkInt <= Android.OS.BuildVersionCodes.Q)
+		{
+			authenticateCode = _biometricManager.CanAuthenticate(); // TODO Eliminate warning somehow?
```

Comment:
Should this todo still be relevent?

Reply:
Yes, I don't know why we need to check the version. This should be done in the suggestion I made on the .NET backlog.

Reply:
Right, if I recall I've added this comment as a reminder to find a way to clear the warning, but I guess we should either remove this comment or reword it to be clearer